### PR TITLE
feat: add Image support to finder

### PR DESCRIPTION
**Pull request by wanglikun7342**
**Description**
add Image Finder to support a widget use ImageProvider

**Issues**
https://github.com/flutter/flutter/issues/88760

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*
I add new Finder way to find widget, but will not influence old tests

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by wanglikun7342** [comment]
@jmagman @jonahwilliams please give a review to this pr

**Comment by jmagman** [comment]
Maybe @dkwingsmt? I'm not familiar with the framework/finder code at all...

**Comment by wanglikun7342** [comment]
@goderbauer

File: packages/flutter_test/lib/src/finders.dart [hunk 1]
```diff
@@ -103,6 +103,19 @@ class CommonFinders {
     );
   }
 
+  /// Finds [Image] and [FadeInImage] widgets containing ImageProvider equal to the
```
Comment (dkwingsmt) [inline packages/flutter_test/lib/src/finders.dart:106]:
```suggestion
  /// Finds [Image] and [FadeInImage] widgets containing `image` equal to the
```

File: packages/flutter_test/lib/src/finders.dart [hunk 2]
```diff
@@ -690,6 +703,27 @@ class _WidgetTypeFinder extends MatchFinder {
...
+  final ImageProvider image;
+
+  @override
+  String get description => 'image "$image"';
+
+  @override
+  bool matches(Element candidate) {
+    final Widget widget = candidate.widget;
+    if (widget is Image) {
+      print(widget);
```
Comment (dkwingsmt) [inline packages/flutter_test/lib/src/finders.dart:718]:
```suggestion
```

File: packages/flutter_test/test/finders_test.dart [hunk 3]
```diff
@@ -7,6 +7,15 @@ import 'package:flutter/rendering.dart';
 import 'package:flutter_test/flutter_test.dart';
 
 void main() {
+  group('image', () {
+    testWidgets('finds Image widgets', (WidgetTester tester) async {
+      await tester.pumpWidget(_boilerplate(
+        const Image(image: _TestAssetImageProvider('test')),
```
Comment (dkwingsmt) [inline packages/flutter_test/test/finders_test.dart:13]: Does it mean, if a developer would like to unit test a widget tree that contains `AssetImage`, they would always need to replace the `AssetImage` into something like `_TestAssetImageProvider`? If that's the case, this method is close to unusable.

**Review by dkwingsmt — CHANGES_REQUESTED** [review]

**Comment by wanglikun7342** [comment]
@dkwingsmt I change the Test to FileImage

**Review by dkwingsmt — APPROVED** [review]
LGTM

**Pull request merged**